js/src/hooks/useMCProductStatistics.js

```diff
@@ -1,15 +1,60 @@
+/**
+ * External dependencies
+ */
+import { useEffect } from '@wordpress/element';
+
 /**
  * Internal dependencies
  */
-import useAppSelectDispatch from '.~/hooks/useAppSelectDispatch';
+import useAppSelectDispatch from './useAppSelectDispatch';
+import useCountdown from './useCountdown';
+import { useAppDispatch } from '.~/data';
 
 /**
  * Call `useAppSelectDispatch` with `"getMCProductStatistics"`.
+ * If the background process is loading, start a countdown and invalidate  the resolution for getMCProductStatistics and getMCProductFeed.
  *
  * @return {useAppSelectDispatch} Result of useAppSelectDispatch.
  */
 const useMCProductStatistics = () => {
-	return useAppSelectDispatch( 'getMCProductStatistics' );
+	const { second, callCount, startCountdown } = useCountdown();
+	const { invalidateResolutionForStoreSelector } = useAppDispatch();
+	const { data, hasFinishedResolution, invalidateResolution, ...rest } =
+		useAppSelectDispatch( 'getMCProductStatistics' );
+
+	const isLoading = hasFinishedResolution && data?.loading ? true : false;
+	const hasStats = hasFinishedResolution && ! data?.loading ? true : false;
```

Comment:
Thanks for the suggestion, I changed a now we are checking if `data.statistics` has data. Also, I changed the response from the API, so instead of an empty array, it returns null for the statistics. See here: https://github.com/woocommerce/google-listings-and-ads/pull/2318/commits/03ceb657f447220226c33bda30999b28a779dc89